### PR TITLE
Support exists query for API key query

### PR DESCRIPTION
**Pull request by ywangd**
Exists query is useful for testing whether expiration date is configured
for an API key or whether a metadata field exists. This PR adds support
for it.

### CODE REVIEW

**Review by tvernum — APPROVED** [review]

**Comment by tvernum** [comment]
Grrr. I had a comment, but it got lost. Give me a second to rewrite it.

File: x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/QueryApiKeyIT.java [hunk 1]
```diff
@@ -367,6 +367,97 @@ public void testSort() throws IOException {
...
+    public void testExistsQuery() throws IOException, InterruptedException {
+        final String authHeader = randomFrom(API_KEY_ADMIN_AUTH_HEADER, API_KEY_USER_AUTH_HEADER);
+
+        // No expiration
+        createApiKey("test-exists-1", null, null, Map.of("value", 42), authHeader);
+        // A short-lived key
+        createApiKey("test-exists-2", "1ms", null, Map.of("label", "prod"), authHeader);
+        createApiKey("test-exists-3", "1d", null, Map.of("value", 42, "label", "prod"), authHeader);
+        // Ensure the short-lived key is expired
+        Thread.sleep(10);
```
Comment (tvernum) [inline x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/QueryApiKeyIT.java:379]: Would it be possible to run the first 3 queries and then only sleep if <10ms had passed before running the 4th query?

Then we can avoid artificial delays in CI.

Comment (ywangd) [reply x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/QueryApiKeyIT.java:379]: Yeah updated as suggested.

**Comment by ywangd** [comment]
Failure is unrelated and tracked at https://github.com/elastic/elasticsearch/issues/87203

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Pull request merged**